### PR TITLE
Add `width` option

### LINKED ISSUE #51: New option width
Would be nice if I could say that the box should be 90 characters wide.

### PR DESCRIPTION
**Pull request by Caesarovich**
Hello, here is the awaited `width` option. ( fixes #51 ). This PR comes with a little re-arrangement of the code. I moved all the code responsible for determining how to resize/shrink/disable options into a function `determineDimensions`.  From now on, code that handles resizing, shrinking, etc... Should be place in that function.

Features:
- The width option allows to set a fixed size for the box. This is useful in the cases where a user wants a box to be rendered regardless of current terminal dimensions.
- Bug fix: padding overflow. If there wasn't enough space to fit the text (at least one character per line) inside the box, padding would overflow on the side. (This issue has never come before because of the low probability of a padding to be larger than `terminalColumns - bordersWidth - 1`)

Thanks for reviewing !

### CODE REVIEW

**Comment by Caesarovich** [comment]
I'm too used to use it my bad.

File: index.d.ts [hunk 1]
```diff
@@ -176,6 +176,19 @@ export interface Options {
 	```
 	*/
 	readonly titleAlignment?: 'left' | 'right' | 'center';
+
+	/**
+	Set a fix width for the box. (This disables terminal overflow safeties !)
```
Comment (sindresorhus) [inline index.d.ts:181]:
```suggestion
	Set a fixed width for the box.
	
	*Note:* This disables terminal overflow handling and may cause the box to look broken if the user's terminal is not wide enough.
```

File: index.d.ts [hunk 2]
```diff
@@ -176,6 +176,19 @@ export interface Options {
...
+
+	@example
+	```
+	console.log(boxen('foo bar', {width: 15}));
+	// ┌─────────────┐
+	// │foo bar      │
+	// └─────────────┘
+
```
Comment (sindresorhus) [inline index.d.ts:189]:
```suggestion
	```
```

File: index.d.ts [hunk 3]
```diff
@@ -176,6 +176,19 @@ export interface Options {
 	```
 	*/
 	readonly titleAlignment?: 'left' | 'right' | 'center';
+
+	/**
+	Set a fix width for the box. (This disables terminal overflow safeties !)
+
+	@example
+	```
```
Comment (sindresorhus) [inline index.d.ts:184]:
```suggestion
	```
	import boxen from 'boxen';
```

File: index.js [hunk 4]
```diff
@@ -221,6 +221,59 @@ const boxContent = (content, contentWidth, options) => {
...
+		options.width = Math.min(options.width, columns - BORDERS_WIDTH - options.margin.left - options.margin.right);
+	}
+
+	if (options.width - (options.padding.left + options.padding.right) <= 0) {
+		options.padding.left = 0;
+		options.padding.right = 0;
+	}
+
+	return options;
+};
```
Comment (sindresorhus) [inline index.js:275]: Would be good with a few short code comments inside here explaining some of the logic.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js (hunk 4 above)
Comment (Caesarovich) [reply index.js:275]: Very well

**Comment by Caesarovich** [comment]
Very well, you're much better than me at making meaningful phrases ^^'

**Pull request merged**